packages/jupyter-ai/src/components/chat-input.tsx

```diff
@@ -52,36 +179,96 @@ export function ChatInput(props: ChatInputProps): JSX.Element {
 
   return (
     <Box sx={props.sx}>
-      <Box sx={{ display: 'flex' }}>
-        <TextField
-          value={props.value}
-          onChange={e => props.onChange(e.target.value)}
-          fullWidth
-          variant="outlined"
-          multiline
-          onKeyDown={handleKeyDown}
-          placeholder="Ask Jupyternaut"
-          InputProps={{
-            endAdornment: (
-              <InputAdornment position="end">
-                <IconButton
-                  size="small"
-                  color="primary"
-                  onClick={props.onSend}
-                  disabled={!props.value.trim().length}
-                  title="Send message (SHIFT+ENTER)"
-                >
-                  <SendIcon />
-                </IconButton>
-              </InputAdornment>
-            )
-          }}
-          FormHelperTextProps={{
-            sx: { marginLeft: 'auto', marginRight: 0 }
-          }}
-          helperText={props.value.length > 2 ? helperText : ' '}
-        />
-      </Box>
+      <Autocomplete
+        autoHighlight
+        freeSolo
+        inputValue={props.value}
+        onInputChange={(_, newValue: string) => {
+          props.onChange(newValue);
+        }}
+        onHighlightChange={
+          /**
+           * On highlight change: set `highlighted` to whether an option is
+           * highlighted by the user.
+           *
+           * This isn't called when an option is selected for some reason, so we
+           * need to call `setHighlighted(false)` in `onClose()`.
+           */
+          (_, highlightedOption) => {
+            setHighlighted(!!highlightedOption);
+          }
+        }
+        onClose={
+          /**
+           * On close: set `highlighted` to `false` and close the popup by
+           * setting `open` to `false`.
+           */
+          () => {
+            setHighlighted(false);
+            setOpen(false);
+          }
+        }
+        // set this to an empty string to prevent the last selected slash
+        // command from being shown in blue
+        value=""
+        open={open}
+        options={slashCommandOptions}
+        // hide default extra right padding in the text field
+        disableClearable
+        // ensure the autocomplete popup always renders on top
+        componentsProps={{
+          popper: {
+            placement: 'top'
+          },
+          paper: {
+            sx: {
+              border: '1px solid lightgray'
```

Comment:
Could we use JupyterLab CSS variables here, to allow for better appearance in custom themes?